include/aspect/mesh_refinement/isosurfaces.h

```diff
@@ -0,0 +1,146 @@
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef _aspect_mesh_refinement_isosurfaces_h
+#define _aspect_mesh_refinement_isosurfaces_h
+
+#include <aspect/mesh_refinement/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+#include <deal.II/base/parsed_function.h>
+
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    namespace internal
+    {
+      enum class PropertyType
+      {
+        Temperature,
+        Composition
+      };
+
+      struct Property
+      {
+        public:
+          /**
+           * Constructor. Converts a property name into a structure containing a property type
+           * and an index. If the property contains multiple items (e.g. the property compositional field has a
+           * field index) the index refering to the particluar item of that proparty is stored in the variable index.
+           * @param property_name The name of a property, which can be Temperature for the temperature field or
+           * the name of a compositional field listed in the parameter available_compositions.
+           * @param available_compositions A list of names of the available compositional fields.
+           */
+          Property(const std::string &property_name,
+                   const std::vector<std::string> &available_compositions);
+
+          /**
+           * The Property type of the property
+           */
+          PropertyType type;
+
+
+          /**
+           * An index, in case the property type contains multiple values. This is
+           * currently only used for storing which compositional field the property
+           * corresponds to.
+           */
+          unsigned int index;
+      };
+
+      struct Isosurface
+      {
+        public:
+          /**
+           * Checks whether the provided values are between the min and max value for the
+           * set properties of the isosurface. This function assumes that the order of the
+           * provided @p values matches the order in which the properties are stored.
+           *
+           * This function assumes than @p values and all the vectors in isosurfaces already
+           * have the same length.
+           */
+          bool are_all_values_in_range(const std::vector<double> &values) const;
+
+          std::vector<double> min_values;
+          std::vector<double> max_values;
+          int min_refinement;
```

Comment:
Yes, it could be, but the cell->level() is an int (I don't know why), so it will give warnings. I could cast it to an int every time it is compared with the cell level, but I think this is the more clean solution.